### ISSUE TITLE
Audio Encoder runs out of memory 

### ISSUE THREAD

**Issue by schoelst** [toniebox-reverse-engineering/teddycloud#139]
First off, thanks of creating TeddyCloud. This project is AMAZING! 

Sadly the Audio Encoder on the website seems to run out of memory sometimes. The following error is then repeated over and over (seems to be stuck in an infinite loop):

```
ERROR|toniefile.c:0344:toniefile_encode| Not enough space in this block
```

This problem occurred reproducibly for some CD rips I tried to convert and it seems to correlate with the total length. For some CDs with up to ~40 min of total play time it worked.

I am running teddycloud release 0.35 in a docker container on a computer with 16Gb RAM. I don't think that the RAM is the problem, because I did not observe that it would run short at any point.

How can I get more insights?
Is there a way to convert the audio locally for debugging and performance? - I am on MacOS, so only a command line version of TeddyBench would be an option.

**Comment by schoelst** [comment]
I tested around a little more and the problem seem to be individual songs. For these the encoding process runs into the error above a bunch of times. The error corresponds to the following case the following lines of code: https://github.com/toniebox-reverse-engineering/teddycloud/blob/21f9c08e3daa13dde35991f7f568674f2e98641f/src/toniefile.c#L342-L346

I am happy to help finding a fix, but sadly don't understand the code well enough yet.

**Comment by Freneato** [comment]
i have the same issue - see in the forum
[https://forum.revvox.de/t/error-uploading-new-content-via-teddycloud-not-enough-space-in-this-block/134/6?u=tomyknock](url)

**Comment by SciLor** [comment]
You may try a different bitrate for those files, which is settable with v0.4.1

**Comment by SciLor** [comment]
As an alternative I have provided a possible fix with commit
the build will be available on the develop tag as soon as it is build.

**Comment by schoelst** [comment]
Sorry for the late reply, was on vacation and could not check. The issue is solved for me in `v0.4.3`. Thanks!
